src/vs/workbench/contrib/files/browser/files.contribution.ts

```diff
@@ -362,6 +362,15 @@ configurationRegistry.registerConfiguration({
 			'description': nls.localize('fileDialogDefaultPath', "Default path for file dialogs, overriding user's home path. Only used in the absence of a context-specific path, such as most recently opened file or folder."),
 			'scope': ConfigurationScope.MACHINE
 		},
+		'files.dialog.openFolderStartLocation': {
+			'type': 'string',
+			'enum': [
```

Comment:
Do we envision any more entries? Why not a simple `boolean` for this?

Reply:
I thought it was clear to have the name + value of the setting with an enum. If we do a boolean, then the setting name will be something like "files.dialog.startInFolderParent".